src/clerk.rs

```diff
@@ -0,0 +1,145 @@
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use crate::config::Config;
+use clerk_rs::{
+    apis::{sessions_api, users_api},
+    clerk::Clerk,
+    models::{self, UpdateUserMetadataRequest},
+    ClerkConfiguration,
+};
+use secrecy::ExposeSecret;
+use serde_json::json;
+
+#[derive(Debug)]
+pub enum ClerkError {
+    InvalidSession,
+    FailedToGetUser,
+    FailedToUpdateUser,
+    Other(String), // For other errors, if needed
+}
+
+impl std::fmt::Display for ClerkError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ClerkError::InvalidSession => write!(f, "Invalid session"),
+            ClerkError::FailedToGetUser => write!(f, "Failed to get user"),
+            ClerkError::FailedToUpdateUser => write!(f, "Failed to update user"),
+            ClerkError::Other(msg) => write!(f, "{}", msg),
+        }
+    }
+}
+
+impl std::error::Error for ClerkError {}
+
+pub struct ClerkResponse {
+    pub user: models::User,
+    pub session: models::Session,
+}
+
+pub struct ClerkHandler {
+    pub client: Clerk,
+    pub dispense_limit_interval: u64,
+}
+
+impl ClerkHandler {
+    // Initialize a new ClerkClientHandler with Clerk configuration
+    pub fn new(config: &Config) -> Self {
+        let clerk_secret_key = config
+            .clerk_secret_key
+            .clone()
+            .expect("Clerk secret key is required");
+        let clerk_key = Some(clerk_secret_key.expose_secret().clone());
+        let clerk_config = ClerkConfiguration::new(None, None, clerk_key, None);
+        let client = Clerk::new(clerk_config);
+        ClerkHandler {
+            client,
+            dispense_limit_interval: config.dispense_limit_interval,
+        }
+    }
+
+    pub async fn update_user_claim(&self, user_id: &str) -> Result<models::User, ClerkError> {
+        let timestamp = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards");
+        let user = self.get_user(user_id).await?;
+        let update_request = Some(UpdateUserMetadataRequest {
+            public_metadata: None,
+            unsafe_metadata: None,
+            private_metadata: Some(json!({
+                "claim_timestamp":  timestamp.as_secs().to_string(),
```

Comment:
I think it is still better to store this information on our side(we already do that with `DispenseTracker`). If something doesn't work correctly, we can always restart the server and wipe the data. It allows test fixes or investigate different problems.